.github/workflows/cherry-pick.yml

```
@@ -2,126 +2,104 @@
 # Licensed under the MIT License.
 
 # This action cherry-picks CVE fix commits from fast-track branches to corresponding
-# development branches for each version of Mariner.
-# To use this action, create a comment in a merged PR to any 'fasttrack/*' branch:
-#   /cherry-pick <target-branch-1>, <target-branch-2>, etc
-# The author of the comment must be a member or owner of the repository for the action
-# to work.
+# development branches for each version of Mariner. By default, the workflow is run
+# whenever a CVE fix PR to a fast-track branch is merged. Additional target branches
+# for cherry-pick can be specified by adding a comment to the original PR with the
+# following syntax:
+#   /cherry-pick <target-branch-1> <target-branch-2>
+# Note the list of target branches is space-separated, and the commenter must be a
+# member or owner of the repository for the action to work.
 # After cherry-pick succeeds for a branch, a new PR will be created to merge the commit
 # to that target branch, and a label will be added to the original PR to indicate that
 # the commit has been cherry-picked to that branch.
 name: Cherry pick commits from PR
 on:
-  issue_comment:
+  pull_request:
     types:
-      - created
-jobs:
-  # Basic information extraction and validation
-  pr_info:
-    if: |
-      github.event.issue.pull_request &&
-        startsWith(github.event.comment.body, '/cherry-pick ') &&
-        (github.event.comment.author_association == 'MEMBER' ||
-         github.event.comment.author_association == 'OWNER')
+      - closed
+    branches:
+      - 'fasttrack/*'
+    paths:
+      - 'SPECS/**'
+      - 'SPECS-EXTENDED/**'
+      - 'SPECS-SIGNED/**'
+
+permissions:
+  contents: read
+  pull-requests: read
+defaults:
+  run:
+    shell: bash
 
+env:
+  CHERRY_PICK_BRANCH_MAPPING: '{"fasttrack/1.0": "1.0-dev", "fasttrack/2.0": "main"}'
```

We don't support 1.0, so we can simplify a bit.
```suggestion
  CHERRY_PICK_BRANCH_MAPPING: '{"fasttrack/2.0": "main"}'
```